docs/en/latest/plugins/ai-rag.md

```diff
@@ -0,0 +1,202 @@
+  | encoding_format | No           | string   | The format to return the embeddings in. Can be either `float` or `base64`. Defaults to `float`.                            |
+  | dimensions      | No           | integer  | The number of dimensions the resulting output embeddings should have. Only supported in text-embedding-3 and later models. |
+
+For other parameters please refer to the [Azure OpenAI embeddings documentation](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#embeddings).
+
+- Parameters of `ai_rag.vector_search`
+
+  - Azure AI Search
+
+  | **Field** | **Required** | **Type** | **Description**              |
+  | --------- | ------------ | -------- | ---------------------------- |
+  | fields    | Yes          | String   | Fields for the vector search |
+
+  For other parameters please refer the [Azure AI Search documentation](https://learn.microsoft.com/en-us/rest/api/searchservice/documents/search-post).
+
+Example request body:
+
+```json
+{
+  "ai_rag": {
+    "vector_search": { "fields": "contentVector" },
+    "embeddings": {
+      "input": "which service is good for devops",
+      "dimensions": 1024
+    }
+  }
+}
+```
+
+## Example usage
+
+First initialise these shell variables:
+
+```shell
+ADMIN_API_KEY=edd1c9f034335f136f87ad84b625c8f1
+AZURE_OPENAI_ENDPOINT=https://name.openai.azure.com/openai/deployments/gpt-4o/chat/completions
+VECTOR_SEARCH_ENDPOINT=https://name.search.windows.net/indexes/indexname/docs/search?api-version=2024-07-01
+EMBEDDINGS_ENDPOINT=https://name.openai.azure.com/openai/deployments/text-embedding-3-small/embeddings?api-version=2023-05-15
+EMBEDDINGS_KEY=secret-azure-openai-embeddings-key
+SEARCH_KEY=secret-azureai-search-key
+AZURE_OPENAI_KEY=secret-azure-openai-key
+```
+
+Create a route with the `ai-rag` and `ai-proxy` plugin like so:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes/1" -X PUT \
+  -H "X-API-KEY: ${ADMIN_API_KEY}" \
+  -d '{
+  "uri": "/rag",
+  "plugins": {
+    "ai-rag": {
+      "embeddings_provider": {
+        "azure_openai": {
+          "endpoint": "'"$embeddings_endpoint"'",
+          "api_key": "'"$embeddings_key"'"
+        }
+      },
+      "vector_search_provider": {
+        "azure_ai_search": {
+          "endpoint": "'"$vector_search_endpoint"'",
+          "api_key": "'"$search_key"'"
+        }
+      }
+    },
+    "ai-proxy": {
+      "auth": {
+        "header": {
+          "api-key": "'"$azure_openai_key"'"
+        },
+        "query": {
+          "api-version": "2023-03-15-preview"
+         }
+      },
+      "model": {
+        "provider": "openai",
+        "name": "gpt-4",
+        "options": {
+          "max_tokens": 512,
+          "temperature": 1.0
+        }
+      },
+      "override": {
+        "endpoint": "'"$azure_openai_endpoint"'"
```

Comment:
the variables in the PUT request should be all caps as well to match the env vars above